pkg/services/datasources/service/datasource.go

```diff
@@ -252,6 +295,123 @@ func (s *Service) AddDataSource(ctx context.Context, cmd *datasources.AddDataSou
 	})
 }
 
+// This will valid validate the instance settings return a version that is safe to be saved
+func (s *Service) prepareInstanceSettings(ctx context.Context, pluginContext backend.PluginContext, settings *backend.DataSourceInstanceSettings) (*backend.DataSourceInstanceSettings, error) {
+	operation := backend.AdmissionRequestCreate
```

Comment:
Nit. A bit unclear since this seems to only be used in returned errors. Should it be used more/in other places?

Reply:
The operation?  it is part of the standard admission controller signature -- we are not using it for anything useful yet; but I don't want to deviate from the standard pattern unless we have a good reason